### ISSUE TITLE
Magellan not adding active class ever (maybe due to MediaWiki?)

### ISSUE THREAD

**Issue by Acccent** [foundation/foundation-sites#12064]
Hi all,
I'm trying to implement Foundation 6 with WikiMedia, using the [Foreground skin](https://foreground.wikiproject.net/wiki/Main_Page) as a base (but that uses Foundation 5 so I'm upgrading it).

I have the following issue after initializing the Magellan plugin after DOM load via JS:

## What should happen?

When scrolling the page or clicking the links in the Magellan menu, the links should receive the `is-active` class.

This issue is the most recent similar one: https://github.com/foundation/foundation-sites/pull/11654 but I am not sure it is the same and the issue is marked as merged...

## What happens instead?

No class is added, it seems no event is fired. This is true both for manually scrolling and clicking on the links (clicking does scroll smoothly to the correct position though).

When inspecting the Magellan object, the `scrollPos` property never changes.

## Test Case and/or Steps to Reproduce (for bugs)

Because this is using MediaWiki, I can't completely reproduce it in a testing environment, but I managed to make this CodePen which does show the issue:
https://codepen.io/Acccent/pen/Baowavp

This is the implementation – I can do my best to provide any additional info:

First I find the table of contents that MediaWiki generated and move it to an off-canvas sidebar...
```
toc.appendTo(sidebar).addClass('in-sidebar');
const tocMenu = toc.children('ul');
tocMenu.addClass('vertical menu');
```
Then update the DOM to include relevant data; it's not possible to do it directly in the HTML as MediaWiki generates it in PHP...
```
content.find('.mw-headline').each(function(i) {
    $(this).attr('data-magellan-target', $(this).attr('id'));
});
```
Finally initialize Magellan and Foundation.
```
const tocMagellan = new Foundation.Magellan(tocMenu, { offset: (topBarHeight - 5) });
$(document).foundation();
```

## Your Environment

- Foundation version(s) used: 6.6.2
- Browser(s) name and version(s):
  - Chrome 81.0.4044.129
  - Firefox 75.0
  - Vivaldi 2.11.1811.49
- Device, Operating System and version: Lenovo ThinkPad X1, Windows 10 pro 1909

## Checklist

- [X] I have read and follow the CONTRIBUTING.md document.
- [X] There are no other issues similar to this one. There is one, as linked before, but it was marked as closed and this is slightly different.
- [X] The issue title and template are correctly filled.

**Comment by DanielRuf** [comment]
Hi @Acccent,

it seems you forgot `data-magellan`.

This should work:
https://codepen.io/DanielRuf/pen/zYvErNV
https://codepen.io/DanielRuf/full/zYvErNV

**Comment by DanielRuf** [comment]
Here is your original codepen with just the addition of the `data-magellan` and the correct selector for the body content there: https://codepen.io/DanielRuf/pen/oNjGbVZ

**Comment by Acccent** [comment]
Thank you! The problem wasn't the missing `data-magellan` (which gets added automatically when the plugin is initialized), but me using the wrong selector. The element that's being scrolled is the body, not the inner bodyContent. Silly mistake but I wouldn't have caught it...

Thanks again!

**Comment by Acccent** [comment]
Actually, I'm very confused – looking at my codepen, it seems that selector didn't actually do anything, so I removed that part (and didn't add the `data-magellan` attribute) and it now works in the pen... but still doesn't on my website. I... don't understand.

https://codepen.io/Acccent/pen/Baowavp

**Comment by DanielRuf** [comment]
Well, as it works on the codepen - we can only test where it is broken which is your website. Do you load Foundation and all other scripts at the end of the body like in the codepen?

Try the export button on the bottom right to see the correct structure.

**Comment by Acccent** [comment]
In the process of trying things out to find a solution for this I switched to using a grid frame, and now Magellan doesn't work at all, is this normal? Should I try to make this work or go back to my previous setup?

Here's a demo: https://codepen.io/Acccent/pen/JjYLGMa

**Comment by DanielRuf** [comment]
grid-frame applies styles which might break this (regarding position, overflow, ...). Normally you put another wrapper around that does not have these styles.

For more complex cases you will always need some custom css as we do not cover all possible use cases.

**Comment by Acccent** [comment]
Fair enough! I will continue trying to get this to work and update this thread accordingly.
